src/Exception/InvalidIfResultException.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sofascore\PurgatoryBundle\Exception;
+
+final class InvalidIfResultException extends RuntimeException
```

Comment:
@Brajk19 Maybe:
```suggestion
final class InvalidIfExpressionResultException extends RuntimeException
```

Also, wouldn't it be better to extend [`TypeError`](https://www.php.net/manual/en/class.typeerror.php) here?

```suggestion
final class InvalidIfResultException extends TypeError
```

Or maybe [`UnexpectedValueException`](https://www.php.net/manual/en/class.unexpectedvalueexception.php) here?

```suggestion
final class InvalidIfResultException extends UnexpectedValueException
```

Reply:
@HypeMC https://github.com/sofascore/purgatory-bundle/pull/105/commits/18fa0488b43ab872c67dc5e04fdefc112727fe99